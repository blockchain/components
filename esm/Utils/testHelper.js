import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}