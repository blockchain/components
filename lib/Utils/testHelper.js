"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}