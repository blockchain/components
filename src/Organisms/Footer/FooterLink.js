// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled, { type ReactComponentFunctional } from 'styled-components'

import Link, { type PropsType as LinkPropsType } from '../../Atoms/Links/Link'

export type PropsType = {
  +children?: React.Node,
}

const Wrapper: ReactComponentFunctional<LinkPropsType> = styled(Link).attrs({
  size: '1rem',
  weight: 500,
})`
  position: relative;
  white-space: nowrap;
`

const FooterLink = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterLink.propTypes = {
  children: PropTypes.node.isRequired,
}

export default FooterLink
