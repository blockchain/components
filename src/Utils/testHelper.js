import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
