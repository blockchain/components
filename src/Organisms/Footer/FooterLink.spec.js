import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import FooterLink from './FooterLink'

describe('FooterElement', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<FooterLink>FooterLink</FooterLink>)

    expect(tree).toMatchSnapshot()
  })
})
