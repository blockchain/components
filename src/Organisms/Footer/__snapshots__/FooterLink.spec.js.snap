// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterElement should render correctly 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 500;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  color: #004A7C;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  position: relative;
  white-space: nowrap;
}

<FooterLink>
  <Styled(Link)>
    <Link
      capitalize={false}
      className="c0"
      e2e="defaultLink"
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={500}
    >
      <styled.a
        capitalize={false}
        className="c0"
        e2e="defaultLink"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={500}
      >
        <a
          className="c0 c1"
          data-e2e="defaultLink"
          opacity={1}
          size="1rem"
        >
          FooterLink
        </a>
      </styled.a>
    </Link>
  </Styled(Link)>
</FooterLink>
`;
